### DISCUSSION TITLE
Is there any way to inspect android screen build in Jetpack Compose using Appium inspector?

### DISCUSSION THREAD

**Discussion by attrideepak** [appium/appium#16329] [Q&A] [answered]
Some of the screen in my app are build in Jetpack Compose. I want to inspect these screens using appium inspector, is it possible.

**Comment by KazuCocoa** [comment, marked as answer]
Did you tried out https://github.com/appium/appium-espresso-driver#jetpack-compose-support?

**Reply by attrideepak** [reply]
I tried it and it works but my question here is can I inspect such screens using appium inspector since there is no option to provide a sub driver or do I have to get the page source using driver.getPageSource in a XML .

**Reply by KazuCocoa** [reply]
So, you could not get screenshots/xml page source in the inspector even after changing the driver via settings api as below? (or not yet to provide the settings api?)
https://github.com/appium/appium-espresso-driver#settings-api

<img width="654" alt="image" src="https://user-images.githubusercontent.com/5511591/149057657-2b041105-daad-4317-8e49-57e5443448f2.png">

Then, i need to check it since i haven't tried it via the inspector, or you also could ask it in https://discuss.appium.io/

**Reply by attrideepak** [reply]
Thanks..this helps. It resolved my doubt.
In update settings pass json as `{"driver":"compose"}`

**Reply by vkgodena** [reply]
with {"driver":"compose"} I'm getting this error on my appium inspector Call to 'updateSettings' failed: Setting 'driver' is not supported. Only the following settings are supported: [actionAcknowledgmentTimeout, allowInvisibleElements, ignoreUnimportantViews, elementResponseAttributes, enableMultiWindows, enableNotificationListener, keyInjectionDelay, scrollAcknowledgmentTimeout, shouldUseCompactResponses, waitForIdleTimeout, waitForSelectorTimeout, normalizeTagNames, shutdownOnPowerDisconnect, simpleBoundsCalculation, trackScrollEvents, wakeLockTimeout, serverPort, mjpegServerPort, mjpegServerFramerate, mjpegScalingFactor, mjpegServerScreenshotQuality, mjpegBilinearFiltering, useResourcesForOrientationDetection]. Can you please help me. Thank you

**Reply by KazuCocoa** [reply]
What is your espresso driver version? This is available since version 1.46.0 as https://github.com/appium/appium-espresso-driver#jetpack-compose-support
Btw, UIA2 cannot work with the settings

**Comment by vkgodena** [comment]
I'm using the normal Appium inspector to detect the elements. I'm not
using any espresso driver. using UIAutomator2. As per my understanding
UiAutomator can see the semantics tree and is able to identify the
elements. Am I right?

**Reply by KazuCocoa** [reply]
UIAutomator gets data via AccessibilityNodeInfo, so only compose's data like semantics stuff exposed to the AccessibilityNodeInfo is available via UIA2. Compose's data that is not exposed to the AccessibilityNodeInfo is not available.

**Reply by attrideepak** [reply]
You need to provide capability ` "automationName": "Espresso"` and then  update settings to `{"driver":"compose"}` to inspect composable screens. You will not be able to inspect compose elements without giving desired capability `"automationName": "Espresso"`
